source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+   :caption: src/Controller/RestaurantController.php
+   :language: php
+
+The controller file defines the ``index()`` method, which displays text on
+the web app's front page. The file also defines the ``browse()`` method,
+which finds documents in which the ``borough``
+field is ``'Queens'`` and the ``name`` field contains the string ``'Moon'``.
+This method then displays the documents at the ``/restaurant/browse/`` route. The
+``browse()`` method uses the ``QueryBuilder`` class to construct the query.
+
+Specify Routes
+``````````````
+
+In the ``config/routes.yaml`` file, add the following routes:
+
+.. code-block:: yaml
+   :caption: config/routes.yaml
+
+   # main routes
+   restaurant_index:
+     path: /
+     controller: App\Controller\RestaurantController::index
+     methods: [GET]
+   
+   # browse restaurants
+   restaurant_browse:
+     path: /restaurant/browse
+     controller: App\Controller\RestaurantController::browse
+     methods: [GET]
+
+This file sets the following routes in the application:
+
+- ``restaurant_index [GET]``: landing page
+- ``restaurant_browse [GET]``: retrieves and displays specific restaurants
+
+Customize Templates
+```````````````````
+
+Next, create templates to customize the web app's appearance.
+
+Create the ``templates/restaurant`` directory and populate it with the
+following files:
+
+- ``index.html.twig``
+- ``browse.html.twig``
+
+Paste the following code into the ``index.html.twig`` file:
+
+.. code-block:: html
+   :caption: templates/restaurant/index.html.twig
+
+   {# templates/restaurant/index.html.twig #}
+   
+   {% extends 'base.html.twig' %}
+   
+   {% block body %}
+       <h1>Welcome to the Symfony MongoDB Quickstart!</h1>
+   {% endblock %}
+
+Paste the following code into the ``browse.html.twig`` file:
+
+.. code-block:: html
+   :caption: templates/restaurant/browse.html.twig
+
+   {# templates/restaurant/browse.html.twig #}
+   
+   {% extends 'base.html.twig' %}
+   
+   {% block title %}
+       Search Restaurants
+   {% endblock %}
+   
+   {% block body %}
+       <h1>Search Restaurants</h1>
+           {% for restaurant in restaurants %}
+               <p>
+                   <span style="color:green;"><b>Name: </b>{{ restaurant.name }}</span><br>
+                   <b>Borough:</b> {{ restaurant.borough }}<br>
+                   <b>Cuisine:</b> {{ restaurant.cuisine }}<br>
+               </p>
+           {% endfor %}
+   {% endblock %}
+
+Start your Symfony Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following command from the application root directory to start
+your PHP built-in web server:
+
+.. code-block:: bash
+   
+   symfony server:start
+
+After the server starts, it outputs the following message:
+
+.. code-block:: none
+   :copyable: false
+
+   [OK] Web server listening
+        The Web server is using PHP FPM 8.3.4
+        http://127.0.0.1:8000
+
+Open the URL http://127.0.0.1:8000/restaurant/browse in your web browser.
```

Comment:
Should this be an actual hyperlink? If not, consider code formatting for the URL.

Reply:
In the rendered page it is a hyperlink!